airbyte-db/src/main/java/io/airbyte/db/Databases.java

```diff
@@ -39,21 +42,50 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(Databases.class);
 
+  // The Job Database is initialized by SQL script, which writes a server UUID at the end.
+  // So this database is ready when the server UUID record is present.
+  public static final Function<Database, Boolean> IS_JOB_DATABASE_READY = database -> {
+    try {
+      Optional<String> uuid = ServerUuid.get(database);
+      return uuid.isPresent();
+    } catch (Exception e) {
+      return false;
+    }
+  };
+  public static final Function<Database, Boolean> IS_CONFIG_DATABASE_CONNECTED = database -> {
+    try {
+      return database.query(ctx -> ctx.fetchExists(select().from("information_schema.tables")));
+    } catch (Exception e) {
+      return false;
+    }
+  };
+  public static final Function<Database, Boolean> IS_CONFIG_DATABASE_LOADED_WITH_DATA = database -> {
+    try {
+      return database.query(ctx -> ctx.fetchExists(select().from("airbyte_configs")));
+    } catch (Exception e) {
+      return false;
+    }
+  };
+
   public static Database createPostgresDatabase(String username, String password, String jdbcConnectionString) {
     return createDatabase(username, password, jdbcConnectionString, "org.postgresql.Driver", SQLDialect.POSTGRES);
   }
 
-  public static Database createPostgresDatabaseWithRetry(String username, String password, String jdbcConnectionString) {
+  public static Database createPostgresDatabaseWithRetry(String username,
+                                                         String password,
+                                                         String jdbcConnectionString,
+                                                         Function<Database, Boolean> isDbReady) {
     Database database = null;
 
     while (database == null) {
       LOGGER.warn("Waiting for database to become available...");
 
       try {
         database = createPostgresDatabase(username, password, jdbcConnectionString);
-        Optional<String> uuid = ServerUuid.get(database);
-        if (uuid.isEmpty()) {
-          throw new Exception("Server UUID not available yet!");
+        if (!isDbReady.apply(database)) {
+          LOGGER.info("Database is not ready yet");
```

Comment:
That's an improvement on the messaging!
How can we be even more explicit? 
see this user's thread: https://airbytehq.slack.com/archives/C01MFR03D5W/p1626176068263900

Something like this?
```suggestion
          LOGGER.info("Database is not ready yet. Please wait a moment, it might still be initializing...");
```